app/services/message_templates/template/auto_resolve.rb

```diff
@@ -4,8 +4,12 @@ class MessageTemplates::Template::AutoResolve
   def perform
     return if conversation.account.auto_resolve_message.blank?
 
-    ActiveRecord::Base.transaction do
-      conversation.messages.create!(auto_resolve_message_params)
+    if within_messaging_window?
+      ActiveRecord::Base.transaction do
```

Comment:
we won't need the the transaction in this case i think. since aren't doing multiple object creation